ros/src/camera_feed/camera_feed/camera_feed.py

```diff
@@ -0,0 +1,49 @@
+import cv2
+
+import rclpy
+import rclpy.exceptions
+import rclpy.executors
+import rclpy.logging
+
+from rclpy.node import Node
+
+from sensor_msgs.msg import Image
+from cv_bridge import CvBridge, CvBridgeError
+
+
+class CameraFeed(Node):
+    def __init__(self):
+        super().__init__("camera_feed")
+        self.bridge = CvBridge()
+        self.cap = cv2.VideoCapture(0)
+        self.publisher = self.create_publisher(Image, "/video", 10)
+
+    # gets the camera feed and publishes it to /video topic
+    def run(self):
```

Comment:
This should use a timer callback like the [minimal publisher example node](https://docs.ros.org/en/foxy/Tutorials/Beginner-Client-Libraries/Writing-A-Simple-Py-Publisher-And-Subscriber.html#write-the-publisher-node)

```
self.timer = self.create_timer(timer_period, self.timer_callback)
        self.i = 0

...

    def timer_callback(self):
        msg = String()
        msg.data = 'Hello World: %d' % self.i
        self.publisher_.publish(msg)
        self.get_logger().info('Publishing: "%s"' % msg.data)
        self.i += 1
```

Reply:
is this so the node publishes camera data to /video at a consistent rate. That makes sense I can add that

Reply:
Yes it has an adjustable framerate and if you don't have it based on a timer, it will try to go as fast as it possibly can which will unnecessarily consume a lot of CPU resources